app/common/model/src/main/java/io/syndesis/common/model/integration/step/template/TemplateStepLanguage.java

```diff
@@ -0,0 +1,224 @@
+        return preProcessor.getSymbolSyntaxes();
+    }
+
+    public SymbolSyntax getDefaultSymbolSyntax() {
+        return preProcessor.getSymbolSyntaxes().get(0);
+    }
+
+    public boolean isSymbol(String symbol) {
+        return preProcessor.isMySymbol(symbol);
+    }
+
+    public static Collection<TemplateStepLanguage> values() {
+        List<TemplateStepLanguage> languages = new ArrayList<>();
+        languages.add(MUSTACHE);
+        languages.add(VELOCITY);
+        languages.add(FREEMARKER);
+        return Collections.unmodifiableList(languages);
+    }
+
+    public static TemplateStepLanguage stepLanguage(String language) {
+        if (language == null) {
+            return MUSTACHE;
+        }
+
+        for (TemplateStepLanguage sl : TemplateStepLanguage.values()) {
+            if (sl.name().equalsIgnoreCase(language)) {
+                return sl;
+            }
+        }
+
+        return MUSTACHE;
+    }
+
+    /**
+     * Update the integration by visiting its {@link Flow}s and their {@link Step}s
+     *
+     * @param integration the integration
+     * @return the update integration
+     */
+    public static Integration updateIntegration(Integration integration) {
+        Builder integrationBuilder = integration.builder();
+
+        List<Flow> replacementFlows = new ArrayList<>(integration.getFlows());
+        ListIterator<Flow> flows = replacementFlows.listIterator();
+        while(flows.hasNext()) {
+            Flow flow = flows.next();
+            flows.set(updateFlow(flow));
+        }
+
+        return integrationBuilder
+            .flows(replacementFlows)
+            .build();
+    }
+
+    /**
+     * Update the flow by visiting its {@link Step}s
+     *
+     * @param flow the flow
+     * @return the newly updated flow
+     */
+    public static Flow updateFlow(Flow flow) {
+        Flow.Builder flowBuilder = flow.builder();
+
+        List<Step> replacementSteps = new ArrayList<>(flow.getSteps());
+        ListIterator<Step> steps = replacementSteps.listIterator();
+
+        while (steps.hasNext()) {
+            Step source = steps.next();
+            steps.set(updateStep(source));
+        }
+
+        return flowBuilder.steps(replacementSteps).build();
+    }
+
+    /**
+     * Update the template step {@link Step} with required dependencies
+     *
+     * @param step the template step
+     * @return the new step
+     */
+    public static Step updateStep(Step step) {
+        if (StepKind.template != step.getStepKind()) {
+            return step;
+        }
+
+        Map<String, String> properties = step.getConfiguredProperties();
+        TemplateStepLanguage language = stepLanguage(
+                                                                    properties.get(LANGUAGE_PROPERTY));
+
+        Dependency dependency = Dependency.maven(language.mavenDependency());
+        return step.builder()
+                     .dependencies(Collections.singleton(dependency))
```

Comment:
maybe here you can use `addDependency` as it will add dependencies to those already set on the step, unless the goal is to always replace them.

Reply:
They do need replacing as they are mutually exclusive. For example, if I change the language syntax of an existing step then I want to the language dependency replaced completely.